filebeat/scripts/generator/dashboards.go

```diff
@@ -0,0 +1,89 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package generator
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/paths"
+)
+
+const (
+	dashboardEntry = `- id: %s
+  file: %s
+`
+)
+
+func AddModuleDashboard(beatName, module, kibanaVersion, dashboardID string, dashboard common.MapStr, suffix string) error {
+	version, err := common.NewVersion(kibanaVersion)
+	if err != nil {
+		return err
+	}
+	if version.Major < 6 {
+		return fmt.Errorf("saving exported dashboards is not available for Kibana version '%s'", version.String())
+	}
+
+	modulePath := filepath.Join(paths.Resolve(paths.Home, "module"), module)
+	stat, err := os.Stat(modulePath)
+	if err != nil || !stat.IsDir() {
+		return fmt.Errorf("no such module: %s\n", modulePath)
+	}
+
+	dashboardFile := strings.Title(beatName) + "-" + module + "-" + suffix + ".json"
```

Comment:
Why 'title'? Given default case insensitivity Windows/macOS filesystems can be a pain with in comparison to Linux.

Reply:
I thought this format is the name convention for Filebeat module dashboards, as 99% percent of them are named Filebeat-{modulename}-{suffix}.json.